Client/Frontend/Browser/BackForwardListViewController.swift

```diff
@@ -24,12 +24,34 @@ class BackForwardListViewController: UIViewController, UITableViewDataSource, UI
     private let BackForwardListCellIdentifier = "BackForwardListViewController"
     private var profile: Profile
     private lazy var sites = [String: Site]()
-    private lazy var tableView = UITableView()
     private var isPrivate: Bool
     private var dismissing = false
+    private var currentRow = 0
+    lazy var tableView = UITableView()
+    
+    lazy var shadow: UIView = {
+        let shadow = UIView()
+        shadow.backgroundColor = UIColor(colorLiteralRed: 0, green: 0, blue: 0, alpha: 0.2)
+        return shadow
+    }()
+    
     var tabManager: TabManager!
+    var bvc: BrowserViewController!
```

Comment:
I'm wary of passing a reference to the BVC and retaining it in this `BackForwardListViewController`. From an ownership perspective, I can see the BVC's navigation controller to own a reference to this but I don't expect this to retain one back. I would make this a `weak` ref at the very least.